appsmithorg/appsmith
# fix: Add current index property to min max fields

Fixes #20231 

- It is a new feature (non-breaking change which adds functionality)

This pull request adds support for currentIndex/currentRow properties for editable columns in a table widget. This property has been specifically added for min/max/regex/Error Message/Required properties of a column.

This change has been made to achieve consistency in the implementation and API of a table widget column for various properties that it has.

This pull request has a migration which has been unit tested by me. Cypress tests for the feature have also been added.

## Test Plan
> https://github.com/appsmithorg/TestSmith/issues/2380

## Checklist:
### Dev activity


### QA activity:



Testing Plan - by developer

1. Make a column editable.
2. Check `currentIndex` and `currentRow` are available for checking validations in min/max/errorMessage/regex/required fields.
3. Since this PR also involves a migration, please check existing values in min/max/errorMessage/regex/required fields don't break after migration. Existing values in validations can include validations values of type string, boolean and other data type possible, like `{currentIndex == 1}`
